[docs] Add docs on how to set a REST timeout (#7373)
apollographql/apollo-server: docs/source/data/fetching-rest.mdx

}
```

## Setting fetch options

> The second parameter to each of the REST methods is the request options object. This object includes some of the common options you might typically pass to [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/fetch#syntax), including `method`, `headers`, `body`, and `signal`. If you are looking for other advanced options that you cannot find in the Apollo docs, you can set them here and refer to your Fetch API docs.

Reviewer: ```suggestion
The second parameter passed to each REST method is an object containing request options. These include options commonly passed to [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/fetch#syntax), including `method`, `headers`, `body`, and `signal`.

> If you're looking for other advanced options that aren't covered in the Apollo docs, you can set them here and refer to your Fetch API docs.
```

> When you make a call to one of the REST methods, the second parameter has some predefined TypeScript types provided by `@apollo/datasource-rest`, but ultimately it is the same `options` object that is passed to your [fetch implementation](https://developer.mozilla.org/en-US/docs/Web/API/fetch#syntax). If you are looking for other advanced options that you can not find in the Apollo docs, you can set them here and refer to your Fetch API docs.

Reviewer 2: ```suggestion
The second parameter to each of the REST methods is the request options object. This object includes some of the common options you might typically pass to [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/fetch#syntax), including `method`, `headers`, `body`, and `signal`. If you are looking for other advanced options that you cannot find in the Apollo docs, you can set them here and refer to your Fetch API docs.
```

```
this.get('/movies/1', options);
```

### Setting timeouts

> To set a `fetch` timeout, you need to provide an [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) via the `signal` option which will allow you to abort the request with custom logic.

Reviewer: ```suggestion
To set a `fetch` timeout, provide an [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) via the `signal` option, which enables you to abort the request with custom logic.
```

> To set a timeout with a fetch implementation you need to set the `signal` option to an [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) which will allow you to abort the request with custom logic.

Reviewer 2: ```suggestion
To set a `fetch` timeout, you need to provide an [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) via the `signal` option which will allow you to abort the request with custom logic.
```

> Here is an example of a simple timeout after a fixed time for every request:
> ```

Reviewer: ````suggestion
Here's an example of a simple timeout after a fixed time for every request:

```
````

> this.get('/movies/1', {signal: AbortSignal.timeout(myTimeoutMilliseconds)});

Reviewer 2: ```suggestion
this.get('/movies/1', { signal: AbortSignal.timeout(myTimeoutMilliseconds) });
```
